syft/frameworks/torch/mpc/primitives.py

```diff
@@ -0,0 +1,160 @@
+                a some point and then re-access the keys.
+        """
+        type_op = {"eq": "fss_eq", "comp": "fss_comp", "xor_add": "xor_add_couple"}[type_op]
+        primitive_stack = getattr(self, type_op)
+
+        available_instances = len(primitive_stack[0]) if len(primitive_stack) > 0 else -1
+        if available_instances >= n_instances:
+            keys = []
+            for i, prim in enumerate(primitive_stack):
+                # We're selecting on the last dimension of the tensor
+                # [:] ~ [slice(None)]
+                # [:1] ~ [slice(1)]
+                # [1:] ~ [slice(1, None)]
+                # [:, :, :1] ~ [slice(None)] * 2 + [slice(1)]
+                n_dim = len(prim.shape)
+                get_slice = [slice(None)] * (n_dim - 1) + [slice(n_instances)]
+                remaining_slice = [slice(None)] * (n_dim - 1) + [slice(n_instances, None)]
+
+                keys.append(prim[get_slice])
+                if remove:
+                    primitive_stack[i] = prim[remaining_slice]
+
+            return keys
+        else:
+            raise EmptyCryptoPrimitiveStoreError(self, type_op, available_instances, n_instances)
+
+    def provide_primitives(
+        self,
+        crypto_types: Union[str, List[str]],
+        workers: List[AbstractWorker],
+        n_instances: int = 10,
+        **kwargs,
+    ):
+        """
+        Build n_instances of crypto primitives of the different crypto_types given and send them to some workers.
+
+        Args:
+            crypto_types: type of primitive (fss_eq, etc)
+            workers: recipients for those primitive
+            n_instances: how many of them are needed
+            **kwargs: any parameters needs for the primitive builder
+
+        Returns:
+
+        """
+        if isinstance(crypto_types, str):
+            crypto_types = [crypto_types]
+
+        worker_types_primitives = defaultdict(dict)
+        for crypto_type in crypto_types:
+            builder = self._builders[crypto_type]
+
+            primitives = builder(n_party=len(workers), n_instances=n_instances, **kwargs)
+
+            for worker_primitives, worker in zip(primitives, workers):
+                worker_types_primitives[worker][crypto_type] = worker_primitives
+
+        for i, worker in enumerate(workers):
+            worker_message = self._owner.create_worker_command_message(
+                "feed_crypto_primitive_store", None, worker_types_primitives[worker]
+            )
+            self._owner.send_msg(worker_message, worker)
+
+    def add_primitives(self, types_primitives: dict):
+        """
+        Include primitives in the store
+
+        Args:
+            types_primitives: dict {crypto_type: str: primitives: list}
+        """
+        for crypto_type, primitives in types_primitives.items():
+            assert hasattr(self, crypto_type), f"Unknown crypto primitives {crypto_type}"
+
+            current_primitives = getattr(self, crypto_type)
```

Comment:
What about having a dict attribute instead of using getattr?

Reply:
Not sure it makes a big difference right?

Reply:
Not at all, it's just that I don't like `getattr` very much ^^
But after having read it again, I'm fine with it